File: app/src/main/java/com/alternativeinfrastructures/noise/sync/BootStarter.java
package com.alternativeinfrastructures.noise.sync;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.alternativeinfrastructures.noise.sync.bluetooth.BluetoothSyncService;

public class BootStarter extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equalsIgnoreCase(Intent.ACTION_BOOT_COMPLETED))
            BluetoothSyncService.startOrPromptBluetooth(context);
    }
}

File: app/src/main/java/com/alternativeinfrastructures/noise/sync/BootStarter.kt
package com.alternativeinfrastructures.noise.sync

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import com.alternativeinfrastructures.noise.sync.bluetooth.BluetoothSyncService

class BootStarter : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action!!.equals(Intent.ACTION_BOOT_COMPLETED, ignoreCase = true))
            BluetoothSyncService.startOrPromptBluetooth(context)
    }
}
